docs/interface.md

```diff
@@ -0,0 +1,56 @@
+# The Interface
+
+All interactions with the Exercism website are handled automatically. Test runners have the single responsibility of taking a solution, running its tests and returning a standardised output.
+
+## Execution
+
+- A test runner should provide an executable script. You can find more information in the [docker.md](https://github.com/exercism/automated-mentoring-support/blob/master/docs/docker.md) file.
+- The script will receive two parameters:
+  - The slug of the exercise (e.g. `two-fer`).
+  - A path to a directory containing the submitted file(s) (with a trailing slash).
+- The script must write a file to that directory named `report.json`
+
+## Output format
+
+The `report.json` file should be structured as followed:
+
+```json
+{
+  "status": "fail",
+  "tests": [
+    {
+      "name": "Test that the thing works",
+      "status": "fail",
+      "expected": "42",
+      "actual": "123123",
+      "message": "Expected 42 but got 123123"
+    }
+  ]
+}
+```
+
+The following overall statuses are valid:
+- `pass`: All tests passed
+- `fail`: At least one test failed
+- `error`: To be used when the tests did not run correctly (including if there are any skips)
+
+
+The following per-test statuses are valid:
+- `pass`: The test passed
+- `fail`: The test failed
+- `error`: The test errored
+- `skip`: The test was skipped
```

Comment:
We should not have any skipped tests, as the test runner should ensure that all tests are run. That said, it is not wrong to include it here, as we can then detect if a test runner might not have run all tests.

Reply:
Initially I would propose to skip all feature gated tests. Or to mark failing optional tests as skip.